### PR TITLE
feat: simplify log configuration

### PR DESCRIPTION
**Pull request by zhongwencool**
Fixes <issue-or-jira-number>

### <samp>🤖 Generated by Copilot at 1a4428f</samp>

This pull request improves the configuration schema and the logger configuration module for the emqx broker. It renames and reorganizes some fields, adds aliases and a new formatter option, and refactors and tests the file logger settings.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/{ce,ee}/(feat|perf|fix)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] If there should be document changes, a PR to emqx-docs.git is sent, or a jira ticket is created to follow up
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx/src/config/emqx_config_logger.erl [hunk 1]
```diff
@@ -150,30 +150,18 @@ tr_file_handler({HandlerName, SubConf}) ->
...
         config => (log_handler_conf(SubConf))#{
-            type =>
-                case conf_get("rotation.enable", SubConf) of
-                    true -> wrap;
-                    _ -> halt
-                end,
-            file => conf_get("file", SubConf),
-            max_no_files => conf_get("rotation.count", SubConf),
-            max_no_bytes => conf_get("max_size", SubConf)
+            type => wrap,
```
Comment (lafirest) [inline apps/emqx/src/config/emqx_config_logger.erl:153]: what will happen if the old configuration here is `halt`

Comment (zhongwencool) [reply apps/emqx/src/config/emqx_config_logger.erl:153]: rotation(warp) is better for all situtation, we should  force to use `rotation`.

File: apps/emqx_conf/src/emqx_conf_schema.erl [hunk 2]
```diff
@@ -887,50 +891,40 @@ fields("console_handler") ->
     log_handler_common_confs(console);
 fields("log_file_handler") ->
     [
-        {"file",
+        {"sink_to",
```
[the remark this answers is not in the document]
Comment (zhongwencool) [reply apps/emqx_conf/src/emqx_conf_schema.erl]: I has changed this to `to`

File: apps/emqx_conf/src/emqx_conf_schema.erl [hunk 3]
```diff
@@ -1355,3 +1340,19 @@ validator_string_re(Val, RE, Error) ->
...
+        end,
+        fields("log_file_handler")
+    ),
+    HandlersWithoutName = maps:with(FileFields, Conf),
+    HandlersWithName = maps:without(FileFields, Conf),
+    emqx_utils_maps:deep_merge(#{<<"default">> => HandlersWithoutName}, HandlersWithName).
+
+convert_rotation(undefined, _Opts) -> undefined;
+convert_rotation(#{} = Rotation, _Opts) -> maps:get(<<"count">>, Rotation, 10);
+convert_rotation(Count, _Opts) when is_integer(Count) -> Count.
```
[the remark this answers is not in the document]
Comment (zhongwencool) [reply apps/emqx_conf/src/emqx_conf_schema.erl:1358]: It's already limit in type `range(1,128)`, we don't need to verify here.
I also add a test for this limit emqx_conf_schema_tests:log_rotation_count_limit_test/0

**Review by zmstone — APPROVED** [review]

**Pull request merged**